bot/cogs/dm_relay.py

```diff
@@ -0,0 +1,104 @@
+import logging
+from typing import Optional
+
+import discord
+from discord import Color
+from discord.ext import commands
+from discord.ext.commands import Cog
+
+from bot import constants
+from bot.bot import Bot
+from bot.constants import MODERATION_ROLES
+from bot.utils.checks import with_role_check
+from bot.utils.messages import send_attachments
+from bot.utils.webhooks import send_webhook
+
+log = logging.getLogger(__name__)
+
+
+class DMRelay(Cog):
+    """Relay direct messages to and from the bot."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.webhook_id = constants.Webhooks.dm_log
+        self.webhook = None
+        self.bot.loop.create_task(self.fetch_webhook())
+        self.last_dm_user = None
+
+    @commands.command(aliases=("reply",))
+    async def send_dm(self, ctx: commands.Context, member: Optional[discord.Member], *, message: str) -> None:
+        """
+        Allows you to send a DM to a user from the bot.
+
+        If `member` is not provided, it will send to the last user who DM'd the bot.
+
+        This feature should be used extremely sparingly. Use ModMail if you need to have a serious
+        conversation with a user. This is just for responding to extraordinary DMs, having a little
+        fun with users, and telling people they are DMing the wrong bot.
+
+        NOTE: This feature will be removed if it is overused.
+        """
+        if member:
+            await member.send(message)
+            await ctx.message.add_reaction("✅")
+            return
+        elif self.last_dm_user:
+            await self.last_dm_user.send(message)
+            await ctx.message.add_reaction("✅")
+            return
+        else:
+            log.debug("Unable to send a DM to the user.")
+            await ctx.message.add_reaction("❌")
```

Comment:
These may raise exceptions if the user has DMs disabled. Let's catch exceptions and add an ❌ reaction.

Reply:
Solved in the latest commit. I also added caching of user IDs so we can reply to the latest DM sender through restarts.